Added information on timestamp format for time comparison operators (#5657)
segmentio/segment-docs: src/engage/audiences/index.md

> info ""
> Engage supports nested traits, but the Audience builder doesn’t support accessing objects nested in arrays. When you send arrays of objects, they are flattened into strings. As a result, the same conditions that work on strings will work on the array. Within the builder, you can only use string operations like `contains` and `does not contain` to look for individual characters or a set of characters in the flattened array.

> ### Time Comparision

Reviewer: ```suggestion
### Time comparison
```

> You can use various time comparison operators in your Audience definition. Such operators are: before date, after date, within last, within next, before last, after next. Only ISO timestamps can be used with these operators.

Reviewer: ```suggestion
You can use the following time comparison operators in your audience definition: 
- before date
- after date 
- within last
- within next 
- before last
- after next 

Only ISO timestamps can be used with these operators.
```

> If the timestamp is not a valid ISO timestamp (for example, a trailing 'Z' is missing), it will not be processed by Real-time compute. See [Real-time compute compared to batch](docs/engage/audiences/#real-time-compute-compared-to-batch)

Reviewer: ```suggestion
If the timestamp is not a valid ISO timestamp (for example, a trailing `Z` is missing), Segment won't process the audience in real-time. Learn more about [real-time compute compared to batch](docs/engage/audiences/#real-time-compute-compared-to-batch)
```
